core/src/components/searchbar/searchbar.tsx

```diff
@@ -111,6 +111,11 @@ export class Searchbar implements ComponentInterface {
    */
   @Prop() enterkeyhint?: 'enter' | 'done' | 'go' | 'next' | 'previous' | 'search' | 'send';
 
+  /**
+   * If used in a form, set the name of the control, which is submitted with the form data.
+   */
+  @Prop() name?: string;
```

Comment:
We'd like to a follow a similar architecture that we've done here with `ion-radio`: https://github.com/ionic-team/ionic-framework/blob/3bd1d7e1e501ca82830de1b375b38135ab4f6311/core/src/components/radio/radio.tsx#L62

Basically, defaulting the value of `name` to the `inputId`. 

The `inputId` would be a new member of this class:

```ts
private inputId = `ion-searchbar-${searchbarIds++}`;
```

where `let searchbarIds = 0;` is added outside of the search bar class, at the top or bottom of this file.